lib/pi_piper/sysfs/driver.rb

```diff
@@ -0,0 +1,106 @@
+require 'filewatcher'
+
+module PiPiper
+  module Sysfs
+    class Driver < PiPiper::Driver
+
+      GPIO_HIGH = 1
+      GPIO_LOW  = 0
+
+      def initialize
+        @exported_pins = Set.new
+      end
+
+      def close
+        unexport_all
+        @exported_pins.empty? 
+      end
+      
+  # Support GPIO pins
+      def pin_direction(pin, direction)
+        raise ArgumentError, "direction should be :in or :out" unless [:in, :out].include? direction
+        export(pin)
+        raise RuntimeError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(direction_file(pin), direction)
+      end
+
+      def pin_read(pin)
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.read(value_file(pin)).to_i
+      end
+
+      def pin_write(pin, value)
+        raise ArgumentError, "value should be GPIO_HIGH or GPIO_LOW" unless [GPIO_LOW, GPIO_HIGH].include? value
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(value_file(pin), value)
+      end
+
+      def pin_set_pud(pin, value)
+        raise NotImplementedError, "Pull up/down not available with this driver. keep it on :off" unless value == :off
+      end
+      
+      def pin_set_trigger(pin, trigger)
+        raise ArgumentError, "trigger should be :falling, :rising, :both or :none" unless [:falling, :rising, :both, :none].include? trigger
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(edge_file(pin), trigger)
+      end
+
+      def pin_wait_for(pin, trigger)
+        pin_set_trigger(pin, trigger)
+        value = pin_read(pin)
+
```

Comment:
Hi @zsyed91,
I tried to produce what you were speeking about. see the gist I wrote to do it :
https://gist.github.com/elmatou/4fd0edb39ea278c465b211ddbb773f45
While I was watching the cpu usage in another ssh session :
`grep 'cpu ' /proc/stat | awk '{usage=($2+$4)*100/($2+$4+$5)} END {print usage "%"}'`

It happens I always get something around 11.4% more or less....

Of course I'm looking at the edge file, which is not the same thing the value file which cannot have an EOF (it is updated on every clock round).